gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestIntegrationSpec.groovy

```diff
@@ -0,0 +1,247 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.tasks
+
+import com.google.common.collect.ImmutableSet
+import com.palantir.gradle.dist.Serializations
+import com.palantir.gradle.dist.SlsManifest
+import org.gradle.testkit.runner.TaskOutcome
+
+class CreateManifestIntegrationSpec extends AbstractTaskSpec {
+
+    File manifestFile
+
+    def setup() {
+        manifestFile = new File(projectDir, 'build/deployment/manifest.yml')
+    }
+
+    def 'fails if lockfile is not up to date'() {
```

Comment:
instead of creating dependencies (and implicitly testing the pdep discovery) why don't you wire up the input to the task with the pdeps you want. Even better you could refactor the createManfest task so that it can be tested using a unit test since it no longer really relies on gradle details

Reply:
I didn't create any of these tests - I just split them out from the original integration spec and cleaned them up a bit. I also find that integration tests are better for gradle tasks because it runs them in an environment that is closer to how they are actually used. Often the internal logic of a task seems to be fine in an isolated test, but the actual behavior of the task is not what is expected or fails outright when run for real.

Reply:
ok, it seemed like pushing to unit test was a nice way to get rid of the test class hierarchy and speed up tests but its fine this way too